TransferMethod/Sources/UpdateTransferMethodController.swift

```diff
@@ -0,0 +1,474 @@
+
+        let topAnchor = widget.topAnchor.constraint(equalTo: cell.contentView.topAnchor)
+        topAnchor.priority = UILayoutPriority(999)
+
+        let rightAnchor = widget.rightAnchor.constraint(equalTo: cell.contentView.rightAnchor,
+                                                        constant: CGFloat(rightAnchorConstant))
+        rightAnchor.priority = UILayoutPriority(999)
+
+        let bottomAnchor = widget.bottomAnchor.constraint(equalTo: cell.contentView.bottomAnchor)
+        bottomAnchor.priority = UILayoutPriority(999)
+
+        NSLayoutConstraint.activate([leftAnchor, topAnchor, rightAnchor, bottomAnchor])
+
+        return cell
+    }
+}
+
+// MARK: - Presenter - UpdateTransferMethodView -
+extension UpdateTransferMethodController: UpdateTransferMethodView {
+    private func getSectionContainingFocusedField() -> UpdateTransferMethodSectionData? {
+        return presenter.sectionData.first(where: { $0.containsFocusedField == true })
+    }
+
+    private func getSectionIndex(by fieldGroup: String) -> Int? {
+        return presenter.sectionData.firstIndex(where: { $0.fieldGroup == fieldGroup })
+    }
+
+    private func focusField(in section: UpdateTransferMethodSectionData) {
+        section.fieldToBeFocused?.focus()
+        section.reset()
+    }
+
+    func showFooterViewWithUpdatedSectionData(for sections: [UpdateTransferMethodSectionData]) {
+        for section in sections {
+            if let sectionIndex = getSectionIndex(by: section.fieldGroup) {
+                if let footerView = tableView.footerView(forSection: sectionIndex) {
+                    // section is visible, update footer
+                    updateFooterView(footerView, for: sectionIndex)
+                }
+            }
+        }
+
+        //even though the footer is visible, the cell might not be visible. So we need to check if the field that needs
+        // to be focused is visible. We need to scroll to the field in order to focus.
+        if let section = getSectionContainingFocusedField() {
+            let indexPath = getIndexPath(for: section)
+            if isCellVisible(indexPath) {
+                focusField(in: section)
+            } else {
+                isCalledByScrollToRow = true
+                tableView.scrollToRow(at: indexPath, at: .top, animated: true)
+            }
+        }
+    }
+
+    /// Scrollview delegate
+    override public func scrollViewDidEndScrollingAnimation(_ scrollView: UIScrollView) {
+        // update footer once scroll ends
+        if let section = getSectionContainingFocusedField() {
+            if isCellVisible(getIndexPath(for: section)) && isCalledByScrollToRow {
+                focusField(in: section)
+                isCalledByScrollToRow = false
+            }
+        }
+    }
+
+    func fieldValues() -> [(name: String, value: String)] {
+        return widgets.filter { $0.isValueUpdated }.map { (name: $0.name(), value: $0.value()) }
```

Comment:
This only works for Text fields, updating values in drop downs do not work

Reply:
Actually that might be part of my ticket anyway. Will look into it

Reply:
Fixed